### PR TITLE
Handle serde-field that fails to deserialize

### PR DESCRIPTION
**Pull request by jleibs**
### What
arrow2-convert forces us into needing to provide data even if serde fail to deserialize the field. Fall back to default in this case, but specify invalid.

Additionally, this checks to see if a blueprint is totally invalid and if so resets it to the default state. Because this can reset the blueprint immediately upon loading, it was important that this properly set auto-space-views. Needed to pull in a slightly different version of https://github.com/rerun-io/rerun/pull/3077 that still works with the now-required usage of Default.

Resolves: https://github.com/rerun-io/rerun/issues/3127

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3129) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3129)
- [Docs preview](https://rerun.io/preview/8831ec03d85499333648bb15442a39760a2ad41d/docs) 
- [Examples preview](https://rerun.io/preview/8831ec03d85499333648bb15442a39760a2ad41d/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: crates/re_viewport/src/viewport_blueprint.rs [hunk 1]
```diff
@@ -46,8 +47,18 @@ pub struct ViewportBlueprint<'a> {
 }
 
 impl<'a> ViewportBlueprint<'a> {
+    pub fn is_invalid(&self) -> bool {
+        !self.space_views.is_empty()
+            && self
+                .space_views
+                .values()
+                .all(|sv| sv.class_name() == &SpaceViewClassName::invalid())
```
Comment (emilk) [inline crates/re_viewport/src/viewport_blueprint.rs:55]: it's only invalid if _all_ are invalid? What if 9/10 are invalid?

**Review by emilk — APPROVED** [review]

File: crates/re_viewport/src/viewport_blueprint.rs (hunk 1 above)
Comment (jleibs) [reply crates/re_viewport/src/viewport_blueprint.rs:55]: My thought process was that if there are some views that are valid, then we have *something* to show, so I don't want to just throw it away.  The user can always manually click the reset button in that case, or just remove the invalid views to create new ones.

Comment (emilk) [reply crates/re_viewport/src/viewport_blueprint.rs:55]: Fair - let's add that as a comment. Are the invalid views culled somewhere?

Comment (jleibs) [reply crates/re_viewport/src/viewport_blueprint.rs:64]: There's no culling yet, though we could add it. At the moment, instead the layout remains unchanged, but the invalid view just shows a warning message. My thought was that once we get to a point where only some views can be invalid, it's better to make the user aware that something went wrong so they can handle it than to just magically cull them behind the scenes.

**Pull request merged**